### PR TITLE
#139 add delegate info to authcontext

### PR DESCRIPTION
**Pull request by LennartRukower**
Added delegate information to the server and changed the permission type in the frontend accordingly.

### CODE REVIEW

File: server/src/global/auth/authController.ts [hunk 1]
```diff
@@ -127,38 +128,69 @@ export const login = (req: Request, res: Response): void => {
...
+          FROM mitglied_has_evposten
+          LEFT JOIN evposten_has_berechtigung ON mitglied_has_evposten.evposten_evpostenID = evposten_has_berechtigung.evposten_evpostenID
+          WHERE mitglied_has_evposten.mitglied_mitgliedID = ?;
+        `,
+            [result[0].mitgliedID]
+          )
+          .then((directorPermissionsResult: authTypes.DirectorPermissionsQueryResult[]) => {
+            // Adds 'normal' permissions to the role permissions
+            let permissions = [];
+            if (directorPermissionsResult.length !== 0) {
```
Comment (johannesfbraun) [inline server/src/global/auth/authController.ts:161]: The order the permissions are added doesn't seem right to me

Comment (johannesfbraun) [reply server/src/global/auth/authController.ts:161]: A person that has no directorPermissionResult would get no permissions at all

**Review by johannesfbraun — CHANGES_REQUESTED** [review]
See comment

**Review by johannesfbraun — CHANGES_REQUESTED** [review]
LGTM

**Review by johannesfbraun — APPROVED** [review]
LGTM

File: client/src/ts/global/App.tsx [hunk 2]
```diff
@@ -27,9 +27,10 @@ const App: React.FunctionComponent = () => {
       .then((res) => {
         // If the retrieval of the user data is succesfull the user is authenticated
         if (res.status === 200) {
-          const userID = res.data[0].mitgliedID;
-          const userName = res.data[0].name;
-          const permissions = res.data[0].permissions;
+          console.log(res.data.permissions);
```
Comment (qnewald) [inline client/src/ts/global/App.tsx:30]: log statement, remove if unnecessary

**Review by qnewald — APPROVED** [review]
LGTM

**Pull request merged**